src/v/cluster/rm_stm.cc

```diff
@@ -324,7 +324,8 @@ ss::future<checked<model::term_id, tx_errc>> rm_stm::do_begin_tx(
         co_return tx_errc::fenced;
     }
 
-    auto [_, inserted] = _mem_state.expected.emplace(pid, tx_seq);
+    auto [_, inserted] = _mem_state.expected.emplace(
+      pid, std::make_pair(synced_term, tx_seq));
```

Comment:
lets use a struct instead of a tuple, it's easier to use names instead of .first &.second